app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/deployManifestConfig.controller.ts

```diff
@@ -34,21 +30,18 @@ export class KubernetesV2DeployManifestConfigCtrl implements IController {
       this.$scope.stage.moniker,
     ).then((builtCommand: IKubernetesManifestCommandData) => {
       if (this.$scope.stage.isNew) {
-        Object.assign(this.$scope.stage, builtCommand.command);
-        this.$scope.stage.source = this.textSource;
+        defaults(this.$scope.stage, builtCommand.command, {
```

Comment:
`defaults` fixes a small issue when adding a new stage: Currently if a user selects an artifact source in a new patch or deploy stage, then leaves to another stage, and then comes back, the artifact source is overwritten to be a text source. Using defaults provides continuity from any previous edits to this new stage upon revisiting it.